### PR TITLE
add labels to daemonset metadata

### PR DESCRIPTION
**Pull request by ringtail**
Add labels to Daemonset metadata.

### CODE REVIEW

File: deployment/node-problem-detector.yaml [hunk 1]
```diff
@@ -2,6 +2,8 @@ apiVersion: apps/v1
 kind: DaemonSet
 metadata:
   name: node-problem-detector
+  labels:
+    app: node-problem-detector
```
Comment (wangzhen127) [inline deployment/node-problem-detector.yaml]: nit: s/app/k8s-app

**Comment by ringtail** [comment]
@wangzhen127 Hmm. I don't think it is caused by this PR. Could you rerun the test?
```
--- FAIL: TestGoroutineLeak (0.00s)
assertions.go:199: Error Trace:	log_watcher_test.go:194
Error:	Not equal: 9 (expected)
!= 8 (actual)
```

File: deployment/node-problem-detector.yaml (hunk 1 above)
Comment (ringtail) [reply deployment/node-problem-detector.yaml]: I found the label below is app: node-problem-detector. So I think app: node-problem-detector may be more suitable

Comment (wangzhen127) [reply deployment/node-problem-detector.yaml]: Ah, forgot about that. lgtm

**Pull request merged**